### ISSUE TITLE
test_sort failing on s390x

### ISSUE THREAD

**Issue by opoplawski** [gnudatalanguage/gdl#705]
Building current master on Fedora rawhide:
```
test 172
        Start 172: test_sort.pro
172: Test command: /builddir/build/BUILD/gdl-ae9f42c9b514a005cd05ac9156151e7c234426c8/build/testsuite/launchtest "test_sort.pro"
172: Test timeout computed to be: 10000000
172: 05:56:36: Warning: Mismatch between the program and library build versions detected.
172: The library used 3.0 (wchar_t,compiler with C++ ABI 1013,wx containers,compatible with 2.8),
172: and your program used 3.0 (wchar_t,compiler with C++ ABI 1014,wx containers,compatible with 2.8).
172: % Compiled module: TEST_SORT.
172: % Compiled module: BANNER_FOR_TESTSUITE.
172: % Compiled module: GDL_IDL_FL.
172: % TEST_SORT_NELEMENTS: 
172:   NO errors encountered during TEST_SORT_NELEMENTS tests  
172: % Compiled module: ERRORS_CUMUL.
172: % TEST_SORT_NELEMENTS: 
172:   NO errors encountered during TEST_SORT_NELEMENTS tests  
172: % TEST_SORT_NELEMENTS: 
172:   29.0000 errors encountered during TEST_SORT_NELEMENTS tests  
172: % TEST_SORT_NELEMENTS: 
172:   47.0000 errors encountered during TEST_SORT_NELEMENTS tests  
172: corrupted double-linked list
172: Magick: abort due to signal 6 (SIGABRT) "Abort"...
172: TEST EXITED FROM SIGNAL 6
172/201 Test #172: test_sort.pro ......................***Failed    6.01 sec
```

**Comment by opoplawski** [comment]
FWIW - this is still failing with latest git.

**Comment by slayoo** [comment]
Thanks for the report!

**Comment by GillesDuvert** [comment]
Sorry for my lack of knowledge, is this a s390x architecture, as reported in #659? And if yes, what is a s390x architecture? Because if it is not strictly IEEE or EEEI, then the radix sort used in SORT() is not going to work at all and this method should be removed using adequate "ifdef"s . Also, touble will be present in other parts architecture dependent (random numbers and median smoothing)

**Comment by opoplawski** [comment]
s390x is Linux on IBM's System z hardware.  It is a big-endian architecture.  I don't have details on the floating point format, but I think it's pretty standard.

**Comment by GillesDuvert** [comment]
In this case, it could be that this architecture is wrongly detected as little endian (test is in cmake, actually, not in GDL).
But, there is also the `corrupted double-linked list` which stinks...

**Comment by opoplawski** [comment]
This seems to be working now with 1.0.2

**Comment by GillesDuvert** [comment]
As big-endian, probably solved by solving #659 .
